src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs

```diff
@@ -137,6 +137,19 @@ public void Dispose()
             return new NoOpDisposePlugin(lazyTask.Value.Result);
         }
 
+        public void DisposePlugin(IPlugin plugin)
```

Comment:
I think this method should be removed or at least removed from the interface and made internal.